### PR TITLE
fix: check for receipt status during settlement of certificate

### LINKED ISSUE #471: Adding receipt status checking to the `settlement` transaction
## Description

The settlement transaction status should be checked in order to detect if the transaction has been successfully settled.

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

This PR checks the receipt status during the settlement of a certificate.
If the status isn't `success` the certificate is tagged as `InError`.

Fixes #471 

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/agglayer-aggregator-notifier/src/packer/mod.rs [hunk 1]
```diff
@@ -250,3 +270,73 @@ where
...
+    certificate_id: CertificateId,
+    config: Arc<OutboundRpcSettleConfig>,
+) -> Result<Option<TransactionReceipt>, Error>
+where
+    RollupManagerRpc: RollupContract + Settler + Send + Sync + 'static,
+{
+    let receipt = pending_tx
+        .inspect(|tx| info!(hash, "Inspect settle transaction: {:?}", tx))
+        .map_err(|e| {
+            println!("Error: {:?}", e);
```
Comment (hadjiszs) [inline crates/agglayer-aggregator-notifier/src/packer/mod.rs:289]: leftover?

File: crates/agglayer-aggregator-notifier/src/packer/mod.rs [hunk 2]
```diff
@@ -146,76 +151,91 @@ where
...
+                            error: "SettlementTransaction failed due to no receipt status"
+                                .to_string(),
+                        })
+                    }
+                    Some(_) => {
+                        if let Err(error) = state_store.update_certificate_header_status(
+                            &certificate_id,
+                            &CertificateStatus::Settled,
+                        ) {
+                            error!(
```
Comment (atanmarko) [inline crates/agglayer-aggregator-notifier/src/packer/mod.rs:203]: Here we just print error and continue. Is it enough?

**Review by hadjiszs — APPROVED** [review]

File: crates/agglayer-aggregator-notifier/src/packer/mod.rs [hunk 3]
```diff
@@ -146,76 +151,91 @@ where
...
+                            );
+                        }
+                        if let Err(error) = state_store.set_latest_settled_certificate_for_network(
+                            &network_id,
+                            &height,
+                            &certificate_id,
+                            &epoch_number,
+                            &certificate_index,
+                        ) {
+                            error!(
```
Comment (atanmarko) [inline crates/agglayer-aggregator-notifier/src/packer/mod.rs:218]: Same here - have we entered here an inconsistent state, how do we recover?

**Review by atanmarko — APPROVED** [review]

File: crates/agglayer-aggregator-notifier/src/packer/mod.rs (hunk 2 above)
Comment (Freyskeyd) [reply crates/agglayer-aggregator-notifier/src/packer/mod.rs:203]: For now, we can't do much more, unfortunately.
Factorization are planned to address those

**Pull request merged**